### PR TITLE
koordlet: replace int64 with time.Duration in metricsadvisor module

### PR DESCRIPTION
**Pull request by jasonliu747**
### V. Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `make test`

### CODE REVIEW

**Comment by hormes** [comment]
What is the background for this change?

**Comment by jasonliu747** [comment]
> What is the background for this change?

现在关于时间的单位并不一致，有些是int64，有些是time.Duration，之前讨论过会逐渐统一成 time.Duration，也留了一些TODO
https://github.com/koordinator-sh/koordinator/blob/b7df4dddea8d0ff5a4f04fbaa9fc00362da162cc/pkg/koordlet/metricsadvisor/framework/config.go#L46-L54

File: pkg/koordlet/koordlet.go [hunk 1]
```diff
@@ -122,7 +122,7 @@ func NewDaemon(config *config.Configuration) (Daemon, error) {
 
 	collectorService := metricsadvisor.NewMetricAdvisor(config.CollectorConf, statesInformer, metricCache)
 
-	resManagerService := resmanager.NewResManager(config.ResManagerConf, scheme, kubeClient, crdClient, nodeName, statesInformer, metricCache, int64(config.CollectorConf.CollectResUsedIntervalSeconds))
+	resManagerService := resmanager.NewResManager(config.ResManagerConf, scheme, kubeClient, crdClient, nodeName, statesInformer, metricCache, int64(config.CollectorConf.CollectResUsedInterval.Seconds()))
```
[the remark this answers is not in the document]
Comment (jasonliu747) [reply pkg/koordlet/koordlet.go:125]: You are right, but I understood that we never supported the case where the variable CollectResUsedIntervalSeconds is less than or equal to zero.

[the remark this answers is not in the document]
Comment (saintube) [reply pkg/koordlet/koordlet.go:125]: The parameters of the former config are known as intervals in seconds, while with this patch the parameters are changed to general durations. We may add some comments to the arguments' descriptions, or we'd better support the interval of the durations like `2500ms`, `500ms`, etc.

[the remark this answers is not in the document]
Comment (jasonliu747) [reply pkg/koordlet/koordlet.go:125]: added a check to see if the interval is below 1 second.

File: pkg/koordlet/metricsadvisor/framework/config.go [hunk 2]
```diff
@@ -27,28 +27,27 @@ const (
...
 }
 
 func (c *Config) InitFlags(fs *flag.FlagSet) {
-	// todo: replace with DurationVar and modify run feature util
-	fs.IntVar(&c.CollectResUsedIntervalSeconds, "collect-res-used-interval-seconds", c.CollectResUsedIntervalSeconds, "Collect node/pod resource usage interval by seconds")
-	fs.IntVar(&c.CollectNodeCPUInfoIntervalSeconds, "collect-node-cpu-info-interval-seconds", c.CollectNodeCPUInfoIntervalSeconds, "Collect node cpu info interval by seconds")
-	fs.IntVar(&c.CPICollectorIntervalSeconds, "cpi-collector-interval-seconds", c.CPICollectorIntervalSeconds, "Collect cpi interval by seconds")
-	fs.IntVar(&c.PSICollectorIntervalSeconds, "psi-collector-interval-seconds", c.PSICollectorIntervalSeconds, "Collect psi interval by seconds")
-	fs.IntVar(&c.CPICollectorTimeWindowSeconds, "collect-cpi-timewindow-seconds", c.CPICollectorTimeWindowSeconds, "Collect cpi time window by seconds")
+	fs.DurationVar(&c.CollectResUsedInterval, "collect-res-used-interval", c.CollectResUsedInterval, "Collect node/pod resource usage interval. Non-zero values should contain a corresponding time unit (e.g. 1s, 2m, 3h).")
```
Comment (zwzhang0107) [inline pkg/koordlet/metricsadvisor/framework/config.go:48]: please add some explanation like "xxInterval must greater than 1s" for these duration var

Comment (jasonliu747) [reply pkg/koordlet/metricsadvisor/framework/config.go]: done

**Review by jasonliu747 — COMMENTED** [review]
/approve

**Pull request merged**